### PR TITLE
Multipart uploads

### PR DESCRIPTION
**Pull request by kofalt**
This will allow an API consumer to upload a multipart form to send files.
That is, this will now work:

```html
<form action="" method="post" enctype="multipart/form-data">
  <input type="file" name="file" />
</form>
```

Limitation: due to a limitation in the various python libraries I could find, when receiving a form the API will first buffer the stream to disk, rather than parsing as received. This will result in slower processing, since the file is hashed at the end of the upload.

### CODE REVIEW

**Comment by kofalt** [comment]
Update: I tested the tip with a 2 GB and 7 GB file. For both I saw no increased RAM usage in uwsgi.

**Comment by gsfr** [comment]
Needs re-testing for memory consumption and inter-site comms.

File: core.py [hunk 1]
```diff
@@ -285,7 +288,7 @@ def store_file(fd, filename, md5, arcpath, arcname):
             self.abort(404, 'no such ticket')
         arcpath = os.path.join(self.app.config['upload_path'], ticket_id + '.tar')
 
-        if self.request.get('complete').lower() not in ['1', 'true']:
+        if self.request.GET.get('complete', '').lower() not in ('1', 'true'):
```
Comment (kofalt) [inline core.py]: I don't understand this change - is it logically identical? Was it broken before?

Comment (kofalt) [inline core.py]: To clarify, I meant only the `[ ]` -> `( )` change, which was added recently

**Comment by kofalt** [comment]
Confirmed with a 7 GB file that the last two commits do not cause a memory usage explosion, though I imagine depending on the submitted form field order, that may cause extra seeking. Either way, it still works.

Not sure how this could affect inter-site features, since the only user interface that supports inter-site will not be using form uploads. I would expect that if something broke (eg, the GET semantics) it would affect single-site usage equivalently.

File: core.py (hunk 1 above)
Comment (gsfr) [inline core.py]: In this case. ```[]``` and ```()``` are functionally identical. In Python, ```[]``` denotes a mutable array and ```()``` denotes an immutable tuple. The change simply harmonizes the code with the rest of the repo.

**Comment by kofalt** [comment]
Can you confirm what remains before merging? Ref comments above.

**Comment by gsfr** [comment]
I'll test inter-site asap. It's slightly complicated to set up.

**Comment by kofalt** [comment]
Cool, thanks!

**Comment by gsfr** [comment]
Inter-site tests completed with @lmperry's help. Worked fine. Remote uploads are still using excessive memory, not unique to this branch, though. Will address separately.
@kofalt Ready when you are.

**Comment by kofalt** [comment]
Sounds good; merging.

**Pull request merged**